app/javascript/packs/brave_rewards_banner_display.jsx

```diff
@@ -12,20 +12,31 @@ import locale from 'locale/en'
 class BraveRewardsPageForm extends React.Component {
   constructor(props) {
     super(props);
+
+    //check for undefined
+    let social;
+    props.details.socialLinks === undefined ? social = {'twitter': '@', 'youtube': '@', 'twitch': '@'} : social = JSON.parse(props.details.socialLinks);
```

Comment:
So an error would be thrown if `social_links` is set to an empty hash. Going to be slightly nit-picky, but wouldn't this be preferable?

```
    if (Object.keys(props.details.socialLinks).length === 0 && props.details.socialLinks.constructor === Object) {
      social = {'twitter': '@', 'youtube': '@', 'twitch': '@'};
    } else {
      social = JSON.parse(props.details.socialLinks);
    }
```

Reply:
This code is much simpler now that we're parsing in the controller, as you suggested 

There can be two cases here

1. The user has social links saved already, in which case they'll be loaded from props
2. The user doesn't have social links (new user for example) the prop will then be null and will default to '@', '@', '@' for twitter, youtube, and twitch. 

I tested case 2 by clearing the database, and migrating from scratch 
`rake db:reset db:migrate`

I don't think we need to worry about empty hashes `{}` in this case since it'll be undefined if new, and `{'twitter' : '', 'youtube': '', 'twitch' : ''}` if they delete their links.